File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  for (i = 0; i < nums.length; i++) {
    for (j = 0; j < i; j++) {
      if (nums[j] + nums[i] === target) {
        return [i, j]
      }
    }
  }
};

File: 1-two-sum/1-two-sum.rs
use std::collections::HashMap;

impl Solution {
  pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut map = HashMap::new();
    for (i, n) in nums.iter().enumerate() {
      match map.get(n) {
        Some(&v) => return vec![i as i32, v],
        _ => map.insert(target - n, i as i32),
      };
    }
    vec![]
  }
}